### ISSUE TITLE
Dynamic Link: Lightbox Gallery (Triggered from a Single Image or Simple Link)

### ISSUE THREAD

**Issue by black-eye** [elementor/elementor#8054]
## Prerequisites

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**
In WP is quite uneasy to trigger lightbox gallery from a single image or link. This would be a great opportunity for Dynamic Link.

**Describe the solution you'd like**
In current version you can add a dynamic link lightbox for image or video:
![obrazek](https://user-images.githubusercontent.com/5634870/57249590-a4b6ab80-7045-11e9-9991-deb78629fd11.png)

It would be excellent to add a third option for a gallery. It would open the same modal box as for WP native gallery creator. Than you would choose your images and BAM! - gallery from a single image is ready.

**Describe alternatives you've considered**
Some of the galler

**Additional context**

**Comment by TheMangoTrain** [comment]
Your post title is understandable, but your descriptions are confusing.

Here's my take:

**PROBLEM:**
Single images open alone in their own lightbox, although perhaps useful for some special cases, _most end-users expect to see multiple images in a lightbox_ and be able to flow through and browse through multiple images without having to open & close something between each image. 

Every mature lightbox plugin has already dealt with this years ago. You simply have a common class name per-image, and when the user triggers a lightbox, any images with that class name are displayed in the lightbox group. Can't get any simpler than that.

**One solution:**
Currently, when setting up a single image, in the "Link" setting, when you select "Media" you're shown an option to toggle whether it's in a Lightbox. If selecting "No", the link takes you to the media file. If selecting "Yes", the image opens in a Lightbox on it's own (with no other images).

Instead, under the "Link" setting, the media file and lightbox options should be separated, and the options would be "None", "Media file", "Lightbox", "Custom".

The "Media file" option would present options on how you want to handle the Media link. Which file are you linking to... the original, or a resized one? Or are you linking to the actual WordPress post for the media file itself. Another option to specify what browser window to open it in (new tab, same window, etc).

The "Lightbox" option would present an option labeled "Group with Other Images" and another setting defining which lightbox group to be included in. Here you would choose "Entire page" or "New lightbox group: _____". Whatever lightbox group you set, the name is applied to the "Class" of the image element (under "Advanced") e.g. "lightbox_group_bobby". You could operate in reverse too. Go to an image, under "Advanced" and in the class field type "lightbox_group_fruit" and this would work too.

Ultimately, when opening a Lightbox, there would be a scan of the page (or array look up) checking for any other images with the same Lightbox group class as the single image that was clicked.

**Comment by thomasmalley** [comment]
This would be awesome!!

**Comment by SoundYeti** [comment]
This would be So good!

I'm trying to have a gallery of images, and when a single image is opened/clicked it opens a lightbox with a set of dynamic images associated with that image clicked.

**Comment by thomasmalley** [comment]
@SoundYeti This is exactly what I was trying to do too :) 

For Example a gallery of projects, 1 Image per Project to show diversity of work
On click on one of these Images (i.e. Project A) a Lightbox opens showing more images associated with this Project (i.e. more of Project A)

**Comment by black-eye** [comment]
Yes, I meant exactly what @SoundYeti and @thomasmalley described.
@TheMangoTrain's approach is for something different (although also handy).

The difference is, that in my situation, the images in the gallery are not present (I mean visible) somewhere else on the page.

Usecase: just imagine a page with a gallery of galleries. There are several pictures in the grid. When I click on one of them, the lightbox with a "hidden" gallery is opened. In this gallery there are pictures other than those in the main "gallery of galleries".

**Comment by moertel** [comment]
Was anybody here able to find a workaround to achieve this functionality?

**Comment by randomfreeform** [comment]
I TOO want this!!! 
But PLEASE make it work from the ICON widget too! 
Right now ICON can open ONE image or video but not multiple images.

**Comment by kopeliadigital** [comment]
Hi, this plugin solve the problem: https://elementpack.pro/demo/element/advanced-image-gallery/

**Comment by randomfreeform** [comment]
> Hi, this plugin solve the problem: https://elementpack.pro/demo/element/advanced-image-gallery/

Yea for a fee and more plugin bloat.... 
This should be simple for El to do!

**Comment by BarthV** [comment]
I really needthis feature ;(
Any update from EI would be much appreciated !

Thanks.

**Comment by Luxamman** [comment]
Also looking forward to this ASAP.
Thx!

**Comment by Kesiunas** [comment]
Looking forward to these features!

**Comment by makuc** [comment]
Looking forward to them as well.

**Comment by plemaire-activis** [comment]
Also looking forward to this

**Comment by catherinewiltshire** [comment]
Oh this would be a great feature!
A suggested work around until this is added could be creating a popup which is a "lightbox" and a dynamic link on the image to open the popup. It's deffo more work, but if you need this type of function right now it saves adding another plugin!

Obviously the problem with doing it that way is that it won't open to the specific image you clicked on (if using an image as the button)

That's what I'm doing on the site I'm working on until this feature is available anyway!

**Comment by rtpHarry** [comment]
If you have [Unlimited Elements](https://unlimited-elements.com/) then I have posted a custom widget that I've developed using it, which is described here:

  - https://github.com/elementor/elementor/issues/13824#issuecomment-774539887

**Comment by YDM307** [comment]
Try to programme this as well. 

Straight +1

**Comment by tamelander** [comment]
Would be great to have this. Or an explanation of it is possible somehow with custom javascript.

**Comment by Cosirax** [comment]
+1 yes that would be quite awesome to get this soon ! :)

**Comment by tsz1412** [comment]
+1

**Comment by geokongr** [comment]
+1 this is a very useful feature.

**Comment by oooh-boi** [comment]
It's actually very simple to trigger the Lightbox gallery, even to add/append more photos on the fly. I've made a video on that subject so if anyone stumbles upon this topic looking for an easy solution, here it is https://youtu.be/OeWj1IRRoY8

**Comment by otomasz** [comment]
Has this option been finally introduced?

**Comment by djaysan** [comment]
man it's been 5 years.... instead of your AI bloat, you could do better elementor team

**Comment by rtpHarry** [comment]
If you already use Unlimited Elements Pro plugin on your site then the widget I posted might be useful:

https://github.com/elementor/elementor/issues/8054#issuecomment-774540400

Just reposting as I was sure I had made something back in the day but it was hidden for me inside the rolled up comments so I had to dig around a bit.

**Comment by djaysan** [comment]
> If you already use Unlimited Elements Pro plugin on your site then the widget I posted might be useful:
> 
> [#8054 (comment)](https://github.com/elementor/elementor/issues/8054#issuecomment-774540400)
> 
> Just reposting as I was sure I had made something back in the day but it was hidden for me inside the rolled up comments so I had to dig around a bit.

Thanks @rtpHarry - but using another bloated addon plugin defeats the whole purpose of using elementor pro. we shouldn't have to resort to extra plugins for a simple thing like this.
I had to resort to ohhhboi method using an hidden image gallery widget and triggering it through a button. not ideal (i.e not user friendly) but it works and my site is lean with only 5 plugins: elementor + pro, litespeed, rankmath and fail2ban. that's how a website should be: minimal amount of plugins, speed score 99/100 on mobile and desktop.

**Comment by rtpHarry** [comment]
Yeah I agree it should be addressed by Elementor, and you wouldn't install a full plugin like UE just for that one issue. I was just mentioning it as an option if you did already use that addon.
